openapi3.yaml

```diff
@@ -1,57 +1,143 @@
 openapi: 3.0.1
 info:
-  title: service-name
-  description: basic template for map colonies service
+  title: products catalog service
+  description: catalog service
   version: 1.0.0
   license:
     name: MIT
     url: https://opensource.org/licenses/MIT
-
+servers:
+  - url: http://localhost:8080
 paths:
-  /anotherResource:
+  /getProducts:
     get:
-      operationId: getAnotherResource
+      operationId: getProducts
       tags:
-        - anotherResource
-      summary: gets the resource
+        - getProducts
+      summary: get products from db catalog, accroding to the filter in requestBody
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/requestBodyForFilter'          
       responses:
         200:
-          description: OK
+          description: A list of products
           content:
             application/json:
               schema:
-                $ref: '#/components/schemas/anotherResource'
-  /resourceName:
-    get:
-      operationId: getResourceName
+                type: array
+                items:
+                  $ref: '#/components/schemas/product'
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        404:
+          description: NOT FOUND
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
+        500:
+          description: SERVER ERROR
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/error'
```

Comment:
Why do all of these return the same message? Add custom messages for each `status_code`.

Reply:
i didnt mean to return the same error message. just the same schema of error but in the error object i will return different message for each status code. is it make sence?